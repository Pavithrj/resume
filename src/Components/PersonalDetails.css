.personalcontainer {
    text-align: center;
}

.detail {
    background-color: black;
    color: white;
    margin: 0px;
}

.image {
    margin-top: 10px;
    clip-path: circle(100px);
}