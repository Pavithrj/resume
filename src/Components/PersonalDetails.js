import './PersonalDetails.css';
import ProfileContext from './ProfileContext';
import { useContext } from 'react';

function PersonalDetails() {
    const profileData = useContext(ProfileContext);
    console.log('profileData::', profileData);
    const { personalInfo } = profileData;
    const { profilepic, firstName, lastName, designation } = personalInfo;

    return (
        <div className="personalcontainer">
            <div className="detail">Personal details</div>
            <img src={profilepic} alt='img' className='image' />
            <div>{firstName} {lastName}</div>
            <div>{designation}</div>
        </div>
    )
}

export default PersonalDetails;