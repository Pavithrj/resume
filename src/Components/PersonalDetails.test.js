import PersonalDetails from "./PersonalDetails";
import { render } from "@testing-library/react";
import ProfileContext from './ProfileContext';
import profileData from '../Details.json';

describe('PersonalDetails Component', () => {
    const customRender = (ui, { providerProps, ...renderOptions }) => {
        return render(
            <ProfileContext.Provider value={profileData}>{ui}</ProfileContext.Provider>,
            renderOptions,
        )
    }
    it('should print personal details', () => {
        const { asFragment } = customRender(<PersonalDetails />, {
            value: profileData
        });
        expect(asFragment()).toMatchSnapshot();
    })
})