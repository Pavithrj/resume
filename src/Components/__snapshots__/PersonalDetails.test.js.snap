// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalDetails Component should print personal details 1`] = `
<DocumentFragment>
  <div
    class="personalcontainer"
  >
    <div
      class="detail"
    >
      Personal details
    </div>
    <img
      alt="cat"
      class="image"
      src="cat.jpeg"
    />
    <div>
      Akash Badhe
    </div>
    <div>
      Principal Software Engineer (Front End)
    </div>
  </div>
</DocumentFragment>
`;
